### ISSUE TITLE
Gemini Plugin for IntelliJ is Causing Hangs

### ISSUE THREAD

**Issue by bcwhite-code** [GoogleCloudPlatform/cloud-code-intellij#3202]
Every so often, sometimes multiple times per day, IntelliJ hangs with no way to recovery other than "end task" is the task Manager.

JetBrains support confirms that it's a problem with my Google plugin.
https://youtrack.jetbrains.com/issue/IDEA-353776

- IDE type (e.g. IntelliJ, Pycharm): IntelliJ IDEA Ultimate Edition
- IDE version: Build #IU-241.17011.79, built on May 22, 2024
- Gemini Code Assist version:  24.5.1-api-version-223
- Operating System: Windows 10

```
"AWT-EventQueue-0" prio=0 tid=0x0 nid=0x0 waiting on condition
     java.lang.Thread.State: WAITING
 on com.intellij.util.concurrency.Semaphore$Sync@2caafe65
	at java.base@17.0.11/jdk.internal.misc.Unsafe.park(Native Method)
	at java.base@17.0.11/java.util.concurrent.locks.LockSupport.park(LockSupport.java:211)
	at java.base@17.0.11/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:715)
	at java.base@17.0.11/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1047)
	at com.intellij.util.concurrency.Semaphore.waitForUnsafe(Semaphore.java:87)
	at com.intellij.util.concurrency.Semaphore.waitFor(Semaphore.java:79)
	at com.intellij.codeInsight.completion.AsyncCompletion.startThread(CompletionThreading.kt:105)
	at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.startContributorThread(CodeCompletionHandlerBase.java:374)
	at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lambda$scheduleContributorsAfterAsyncCommit$6(CodeCompletionHandlerBase.java:325)
	at com.intellij.codeInsight.completion.CodeCompletionHandlerBase$$Lambda$7543/0x00000001039b8248.accept(Unknown Source)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$safeTransferToEdt$7(NonBlockingReadActionImpl.java:728)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission$$Lambda$2862/0x00000001019b4000.run(Unknown Source)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:209)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:22)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:191)
	at com.intellij.openapi.application.impl.RwLockHolder.runIntendedWriteActionOnCurrentThread(RwLockHolder.kt:204)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:830)
	at com.intellij.openapi.application.impl.ApplicationImpl$2.run(ApplicationImpl.java:419)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithEnabledImplicitRead(RwLockHolder.kt:138)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithImplicitRead(RwLockHolder.kt:129)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWithImplicitRead(ApplicationImpl.java:1152)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:81)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:123)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:43)
	at com.intellij.openapi.application.impl.FlushQueue$$Lambda$551/0x0000000100600d40.run(Unknown Source)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:792)
```

**Comment by ivanporty** [comment]
Thanks for the report @bcwhite-code. We are looking into this

**Comment by ivanporty** [comment]
This should be fixed in the latest release 24.6.1. Please update the plugin and let is know if this works for you or otherwise reopen. Thank you for your report again.

**Comment by imanabu** [comment]
I have 24.6.1, and this still freezes up. Typically this "seems" to happen when IntelliJ is trying to suggest a code completion inline with the editor. I will disable automatic generation for longer... options and see.

**Comment by ivanporty** [comment]
@imanabu thank you for your report. We would like to take a deeper look at this - could you please record a screencast of what is going on with freezes? How bad is the freeze - does it slow down, or hangs?
And, please submit the feedback -> Gemini status bar, Submit feedback with all logs, and add description that this is for GitHub issue #3202 by imanabu.
Thank you!

**Comment by sripasg** [comment]
@imanabu - it would be super helpful to get the feedback using the submit feedback mechanism as described in the previous comment.

**Comment by fgka** [comment]
In my case it renders PyCharm unusable with code complete. To be specific: `Editor > General > Code Complete`. If I also enable `Insert selected suggestion ...` it freezes almost immediately even editing a markdown file.

macOS version (intel): `14.5 (23F79)`

Hardware: `13-inch, 2020, Four Thunderbolt 3 ports`

Plugin version: `24.6.1-IB.1-api-version-223`

PyCharm version:
```text
PyCharm 2024.1.4 (Professional Edition)
Build #PY-241.18034.82, built on June 24, 2024
Licensed to Gustavo Kuhn Andriotti
Subscription is active until September 5, 2024.
Runtime version: 17.0.11+1-b1207.24 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 14.5
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 8
Metal Rendering is ON
Registry:
  debugger.new.tool.window.layout=true
  ide.images.show.chessboard=true
Non-Bundled Plugins:
  String Manipulation (9.14.1)
  de.docs_as_co.intellij.plugin.diagramsnet (0.2.6)
  org.intellij.plugins.hcl (241.17890.8)
  io.kanro.idea.plugin.protobuf (2.0.1)
  name.kropp.intellij.makefile (241.14494.150)
  net.seesharpsoft.intellij.plugins.csv (3.3.0-241)
  com.ibm.appscan.intellij.plugin (2.8)
  com.intellij.bigdatatools.core (241.18034.62)
  com.intellij.bigdatatools.binary.files (241.14494.158)
  PlantUML integration (7.10.1-IJ2023.2)
  com.intellij.bigdatatools.rfs (241.18034.62)
  mobi.hsz.idea.gitignore (4.5.3)
```

**Comment by ivanporty** [comment]
Thanks for the report @fgka! We are sorry about your experience and looking into it.

**Comment by imanabu** [comment]
The Gemini plugin consistently causes IntelliJ to hang still, even after the latest IntelliJ build (IU-241.18034.62) and Gemini version (24.7.1-...233).  The issue occurs immediately when working with Node.js/Firebase functions. The same problem does not happen with VS Code, which runs smoothly with Gemini.

## One Observation

The hang happens most often when I am deleting some characters in a line. It looks like on each delete key hit, there is some code completion updates and that hangs. I have also upped the IDE's memory requirement but that did not help. I am on Windows, and using Java 21.  

I will turn this back on today and will do that feedback method if it happens.

**Comment by imanabu** [comment]
> @imanabu thank you for your report. We would like to take a deeper look at this - could you please record a screencast of what is going on with freezes? How bad is the freeze - does it slow down, or hangs? And, please submit the feedback -> Gemini status bar, Submit feedback with all logs, and add description that this is for GitHub issue #3202 by imanabu. Thank you!

In terms of slow down, no slow down leading to this. It just suddenly, and if it hangs, it will hang the all IntelliJ as if there is a central event loop in all of running IntelliJs. It does not hang other apps. 

Also IJ itself is running still, and consuming CPU and memory and that values changes. It appears that it is the keyboard input that gets disabled and never enabled back again.

**Comment by imanabu** [comment]
Today I have been on the Mac (Intel) and had not had the problem. The issue may be more windows specific, I am guessing.

**Comment by imanabu** [comment]
Day 2 - Still not happening with Intel MacOS. I think that focus on this issue will be on the Windows implementations.

**Comment by Chipintoza** [comment]
it happening every 3-4 Every 3-4 hours on my mac m2 MacOS; webstorm

**Comment by imanabu** [comment]
> it happening every 3-4 Every 3-4 hours on my mac m2 MacOS; webstorm

You are using the latest Webstorm and the Gemini Plug-In? Just confirming.

**Comment by imanabu** [comment]
Switched back to Windows today. I am working on the same Flutter and Firebase Function code base as I did on the Intel Mac. Hope Window freezes today ;-)

**Comment by Chipintoza** [comment]
> > it happening every 3-4 Every 3-4 hours on my mac m2 MacOS; webstorm
> 
> You are using the latest Webstorm and the Gemini Plug-In? Just confirming.

WebStorm 2024.1.5 - and - Gemini Code Assist 24.7.1-api-version-223

Because this happens often... now I have disabled gemini :( and waiting new updates

**Comment by imanabu** [comment]
> > > it happening every 3-4 Every 3-4 hours on my mac m2 MacOS; webstorm
> > 
> > 
> > You are using the latest Webstorm and the Gemini Plug-In? Just confirming.
> 
> WebStorm 2024.1.5 - and - Gemini Code Assist 24.7.1-api-version-223
> 
> Because this happens often... now I have disabled gemini :( and waiting ew updates

I have been editing in Visual Studio Code and refactoring in IntelliJ. Crazy. ;-(

**Comment by imanabu** [comment]
## Great, Just Happened again!

Plenty of memory left in the IDE.

I was hitting DELETE keys (again this is most typical way I can freeze this.) See line 96. 

![image](https://github.com/user-attachments/assets/2ebaa3b0-04c5-448f-958c-1dd06a68f094)

![image](https://github.com/user-attachments/assets/63191b2d-fe7b-45ee-9b9e-a23e03475f54)

**Comment by imanabu** [comment]
> @imanabu thank you for your report. We would like to take a deeper look at this - could you please record a screencast of what is going on with freezes? How bad is the freeze - does it slow down, or hangs? And, please submit the feedback -> Gemini status bar, Submit feedback with all logs, and add description that this is for GitHub issue #3202 by imanabu. Thank you!

##  Feedback Submitted

I have just submitted the official feedback record set to you from the extension.

## Steps To Reproduce

* Generally, when I am editing and start deleting part of a line of a code a freeze will occur. 
* It happens more frequently on Windows 11, the same code base on the Intel Mac did not cause an issue.
* On Windows, killing `cloudcode_cli` (if it's already up, you may not find it) can unfreeze the IDE but after that obviously Gemini Extension no longer works from the IDE. But at least recover from the freeze. _This can be a workaround to save the work before restarting the IDE._

**Comment by ivanporty** [comment]
Thanks for the detailed report @imanabu! There is an underlying issue with this binary that we are working to fix. Will update as soon as it's done.

**Comment by imanabu** [comment]
Been working on Windows 11 today, after gcloud update I made yesrerday, and it has not froze today, Just FYI.

```
Google Cloud SDK 484.0.0
bq 2.1.7
core 2024.07.12
gcloud-crc32c 1.0.0
gsutil 5.30
```

Froze again so this was not correct information.

**Comment by yuseferi** [comment]
same here, Goland

**Comment by xSAVIKx** [comment]
I'm experiencing same behavior across various products and projects. All latest versions with latest plugin version as well.
So far: Webstorm, PyCharm, IDEA - all freeze when you're typing smth in the editor.

Can share a bit of logs from latest PyCharm crash:
```
2024-07-30 11:05:26,876 [76320471]   WARN - duet-ai-language-server - Stream closed, restarting language server
java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: Stream closed
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at com.google.cloud.tools.intellij.languageclient.aicompanion.InlineCompletionResultProvider.getInlineCompletionUsingTriggerMode(InlineCompletionResultProvider.kt:95)
	at com.google.cloud.tools.intellij.languageclient.aicompanion.InlineCompletionResultRenderingService.getInlineCompletions(InlineCompletionResultRenderingService.kt:143)
	at com.google.cloud.tools.intellij.languageclient.aicompanion.InlineCompletionResultRenderingService$sendRequestAndRender$1.invokeSuspend(InlineCompletionResultRenderingService.kt:163)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
Caused by: org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: Stream closed
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:72)
	at org.eclipse.lsp4j.jsonrpc.TracingMessageConsumer.consume(TracingMessageConsumer.java:114)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.request(RemoteEndpoint.java:161)
	at org.eclipse.lsp4j.jsonrpc.services.EndpointProxy.invoke(EndpointProxy.java:91)
	at jdk.proxy3/jdk.proxy3.$Proxy213.sendInlineCompletionRequest(Unknown Source)
	at com.google.cloud.tools.intellij.languageclient.aicompanion.InlineCompletionResultProvider.getInlineCompletionResult(InlineCompletionResultProvider.kt:70)
	at com.google.cloud.tools.intellij.languageclient.aicompanion.InlineCompletionResultProvider.getInlineCompletionUsingTriggerMode(InlineCompletionResultProvider.kt:92)
	... 10 more
Caused by: java.io.IOException: Stream closed
	at java.base/java.lang.ProcessBuilder$NullOutputStream.write(ProcessBuilder.java:445)
	at java.base/java.io.OutputStream.write(OutputStream.java:162)
	at java.base/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:81)
	at java.base/java.io.BufferedOutputStream.flush(BufferedOutputStream.java:142)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:69)
	... 16 more
2024-07-30 11:05:26,876 [76320471]   WARN - org.eclipse.lsp4j.jsonrpc.RemoteEndpoint - Failed to send notification message.
org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: Broken pipe
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:72)
	at org.eclipse.lsp4j.jsonrpc.TracingMessageConsumer.consume(TracingMessageConsumer.java:114)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.notify(RemoteEndpoint.java:126)
	at org.eclipse.lsp4j.jsonrpc.services.EndpointProxy.invoke(EndpointProxy.java:88)
	at jdk.proxy3/jdk.proxy3.$Proxy211.didChange(Unknown Source)
	at com.google.cloud.tools.intellij.languageclient.EditorWatcher$sendDidChangeNotification$1.invokeSuspend(EditorWatcher.kt:307)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
Caused by: java.io.IOException: Broken pipe
	at java.base/java.io.FileOutputStream.writeBytes(Native Method)
	at java.base/java.io.FileOutputStream.write(FileOutputStream.java:349)
	at java.base/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:81)
	at java.base/java.io.BufferedOutputStream.write(BufferedOutputStream.java:122)
	at java.base/java.io.FilterOutputStream.write(FilterOutputStream.java:108)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:68)
	... 13 more
2024-07-30 11:05:26,878 [76320473]   WARN - #c.i.d.PerformanceWatcherImpl - UI was frozen for 33208ms, details saved to /home/USER/.cache/JetBrains/PyCharm2024.1/log/threadDumps-freeze-20240730-110458-PY-241.18034.82-Unsafe.park-33sec
2024-07-30 11:05:26,876 [76320471]   WARN - language-server - Error trying to fetch completions.
java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: Stream closed
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
	at com.google.cloud.tools.intellij.languageclient.LanguageServerCompletionContributor.fillCompletionVariants$lambda$0(LanguageServerCompletionContributor.kt:192)
	at com.intellij.openapi.application.ex.ApplicationUtil.runWithCheckCanceled$lambda$3$lambda$1(ApplicationUtil.kt:57)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:660)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:735)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:691)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:659)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:79)
	at com.intellij.openapi.application.ex.ApplicationUtil.runWithCheckCanceled$lambda$3(ApplicationUtil.kt:54)
	at com.intellij.openapi.application.ex.ApplicationUtil$runWithCheckCanceled$deferred$1.invokeSuspend(ApplicationUtil.kt:74)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
Caused by: org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: Stream closed
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:72)
	at org.eclipse.lsp4j.jsonrpc.TracingMessageConsumer.consume(TracingMessageConsumer.java:114)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.request(RemoteEndpoint.java:161)
	at org.eclipse.lsp4j.jsonrpc.services.EndpointProxy.invoke(EndpointProxy.java:91)
	at jdk.proxy3/jdk.proxy3.$Proxy211.completion(Unknown Source)
	at com.google.cloud.tools.intellij.languageclient.LanguageServerCompletionContributor.fillCompletionVariants$lambda$0(LanguageServerCompletionContributor.kt:188)
	... 16 more
Caused by: java.io.IOException: Stream closed
	at java.base/java.lang.ProcessBuilder$NullOutputStream.write(ProcessBuilder.java:445)
	at java.base/java.io.OutputStream.write(OutputStream.java:162)
	at java.base/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:81)
	at java.base/java.io.BufferedOutputStream.flush(BufferedOutputStream.java:142)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:69)
	... 21 more
```

**Comment by imanabu** [comment]
> I'm experiencing same behavior across various products and projects. All latest versions with latest plugin version as well. So far: Webstorm, PyCharm, IDEA - all freeze when you're typing smth in the editor.
> 
> Can share a bit of logs from latest PyCharm crash:
> 
> ```
> 2024-07-30 11:05:26,876 [76320471]   WARN - duet-ai-language-server - Stream closed, restarting language server
> java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: Stream closed
> 	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
> 	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
> 	at com.google.cloud.tools.intellij.languageclient.aicompanion.InlineCompletionResultProvider.getInlineCompletionUsingTriggerMode(InlineCompletionResultProvider.kt:95)
> 	at com.google.cloud.tools.intellij.languageclient.aicompanion.InlineCompletionResultRenderingService.getInlineCompletions(InlineCompletionResultRenderingService.kt:143)
> 	at com.google.cloud.tools.intellij.languageclient.aicompanion.InlineCompletionResultRenderingService$sendRequestAndRender$1.invokeSuspend(InlineCompletionResultRenderingService.kt:163)
> 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
> 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
> 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
> 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
> Caused by: org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: Stream closed
> 	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:72)
> 	at org.eclipse.lsp4j.jsonrpc.TracingMessageConsumer.consume(TracingMessageConsumer.java:114)
> 	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.request(RemoteEndpoint.java:161)
> 	at org.eclipse.lsp4j.jsonrpc.services.EndpointProxy.invoke(EndpointProxy.java:91)
> 	at jdk.proxy3/jdk.proxy3.$Proxy213.sendInlineCompletionRequest(Unknown Source)
> 	at com.google.cloud.tools.intellij.languageclient.aicompanion.InlineCompletionResultProvider.getInlineCompletionResult(InlineCompletionResultProvider.kt:70)
> 	at com.google.cloud.tools.intellij.languageclient.aicompanion.InlineCompletionResultProvider.getInlineCompletionUsingTriggerMode(InlineCompletionResultProvider.kt:92)
> 	... 10 more
> Caused by: java.io.IOException: Stream closed
> 	at java.base/java.lang.ProcessBuilder$NullOutputStream.write(ProcessBuilder.java:445)
> 	at java.base/java.io.OutputStream.write(OutputStream.java:162)
> 	at java.base/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:81)
> 	at java.base/java.io.BufferedOutputStream.flush(BufferedOutputStream.java:142)
> 	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:69)
> 	... 16 more
> 2024-07-30 11:05:26,876 [76320471]   WARN - org.eclipse.lsp4j.jsonrpc.RemoteEndpoint - Failed to send notification message.
> org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: Broken pipe
> 	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:72)
> 	at org.eclipse.lsp4j.jsonrpc.TracingMessageConsumer.consume(TracingMessageConsumer.java:114)
> 	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.notify(RemoteEndpoint.java:126)
> 	at org.eclipse.lsp4j.jsonrpc.services.EndpointProxy.invoke(EndpointProxy.java:88)
> 	at jdk.proxy3/jdk.proxy3.$Proxy211.didChange(Unknown Source)
> 	at com.google.cloud.tools.intellij.languageclient.EditorWatcher$sendDidChangeNotification$1.invokeSuspend(EditorWatcher.kt:307)
> 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
> 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
> 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
> 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
> Caused by: java.io.IOException: Broken pipe
> 	at java.base/java.io.FileOutputStream.writeBytes(Native Method)
> 	at java.base/java.io.FileOutputStream.write(FileOutputStream.java:349)
> 	at java.base/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:81)
> 	at java.base/java.io.BufferedOutputStream.write(BufferedOutputStream.java:122)
> 	at java.base/java.io.FilterOutputStream.write(FilterOutputStream.java:108)
> 	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:68)
> 	... 13 more
> 2024-07-30 11:05:26,878 [76320473]   WARN - #c.i.d.PerformanceWatcherImpl - UI was frozen for 33208ms, details saved to /home/USER/.cache/JetBrains/PyCharm2024.1/log/threadDumps-freeze-20240730-110458-PY-241.18034.82-Unsafe.park-33sec
> 2024-07-30 11:05:26,876 [76320471]   WARN - language-server - Error trying to fetch completions.
> java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: Stream closed
> 	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
> 	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
> 	at com.google.cloud.tools.intellij.languageclient.LanguageServerCompletionContributor.fillCompletionVariants$lambda$0(LanguageServerCompletionContributor.kt:192)
> 	at com.intellij.openapi.application.ex.ApplicationUtil.runWithCheckCanceled$lambda$3$lambda$1(ApplicationUtil.kt:57)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:660)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:735)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:691)
> 	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:659)
> 	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:79)
> 	at com.intellij.openapi.application.ex.ApplicationUtil.runWithCheckCanceled$lambda$3(ApplicationUtil.kt:54)
> 	at com.intellij.openapi.application.ex.ApplicationUtil$runWithCheckCanceled$deferred$1.invokeSuspend(ApplicationUtil.kt:74)
> 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
> 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
> 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
> 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
> Caused by: org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: Stream closed
> 	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:72)
> 	at org.eclipse.lsp4j.jsonrpc.TracingMessageConsumer.consume(TracingMessageConsumer.java:114)
> 	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.request(RemoteEndpoint.java:161)
> 	at org.eclipse.lsp4j.jsonrpc.services.EndpointProxy.invoke(EndpointProxy.java:91)
> 	at jdk.proxy3/jdk.proxy3.$Proxy211.completion(Unknown Source)
> 	at com.google.cloud.tools.intellij.languageclient.LanguageServerCompletionContributor.fillCompletionVariants$lambda$0(LanguageServerCompletionContributor.kt:188)
> 	... 16 more
> Caused by: java.io.IOException: Stream closed
> 	at java.base/java.lang.ProcessBuilder$NullOutputStream.write(ProcessBuilder.java:445)
> 	at java.base/java.io.OutputStream.write(OutputStream.java:162)
> 	at java.base/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:81)
> 	at java.base/java.io.BufferedOutputStream.flush(BufferedOutputStream.java:142)
> 	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageConsumer.consume(StreamMessageConsumer.java:69)
> 	... 21 more
> ```

Just curious, is this a full crash out of the IDE or IDE freeze? We are tracking about the IDE freeze on this issue using the Gemini feature, so if it is a different behavior than that, that may need to go into another bug report.

It appears that a communication from the IDE to the plug-in hangs and that will hang the whole editor chain based on what I have gathered, and they are working on that issue.

**Comment by alance019** [comment]
I have the same issue with phpstorm on popos.
For me i have to kill phpstorm because there is no interaction possible after the hang...

**Comment by xSAVIKx** [comment]
> Just curious, is this a full crash out of the IDE or IDE freeze? We are tracking about the IDE freeze on this issue using the Gemini feature, so if it is a different behavior than that, that may need to go into another bug report.
> 
> It appears that a communication from the IDE to the plug-in hangs and that will hang the whole editor chain based on what I have gathered, and they are working on that issue.

@imanabu, this is a freeze. The only way to kill the IDE then is to use `pkill` or kill the shell process if you started it from the shell.

**Comment by imanabu** [comment]
> > Just curious, is this a full crash out of the IDE or IDE freeze? We are tracking about the IDE freeze on this issue using the Gemini feature, so if it is a different behavior than that, that may need to go into another bug report.
> > It appears that a communication from the IDE to the plug-in hangs and that will hang the whole editor chain based on what I have gathered, and they are working on that issue.
> 
> @imanabu, this is a freeze. The only way to kill the IDE then is to use `pkill` or kill the shell process if you started it from the shell.

Then, welcome to the club. I think we are all waiting on the fixes being made. The Google team is now well aware of the issue based on previous posts.

**Comment by ivanporty** [comment]
We just released completely reworked Gemini Code Assist functionality in the Cloud Code. It has native IDE completion support, no conflicts with Tab actions, brand new chat UX, chat history and query history, .aiexclude file support (24.8.1 and later). This update has multiple performance and stability improvements and should solve any freezing issues. Please re-open the issue if you see any performance issues.

**Comment by xSAVIKx** [comment]
> We just released completely reworked Gemini Code Assist functionality in the Cloud Code. It has native IDE completion support, no conflicts with Tab actions, brand new chat UX, chat history and query history, .aiexclude file support (24.8.1 and later). This update has multiple performance and stability improvements and should solve any freezing issues. Please re-open the issue if you see any performance issues.

Can confirm that updating to latest IDE (2024.2) + plugin that is split into 2 now fixed the issue for me personally. Thx a lot :+1:

**Comment by imanabu** [comment]
I have been using this on the latest stable IntelliJ Intel Windows for about 8 hours over 3 days and have not experienced any freeze. Also, it previously inserted some trash code during code generation and that isn't happening anymore, been on Flutter DART and TypeScript. Moving to Scala today.
